### PR TITLE
Update Apple Silicon docs

### PR DESCRIPTION
**Pull request by geoffrey4444**
## Proposed changes

The docs for building spectre on Apple Silicon were out of date. This updates the docs. Along the way, I figured out how to get spectre to build with shared libraries, which the instructions now recommend.

Specific changes:
* Removes instructions to build brigand and libsharp, since these are now bundled with spectre
* Fix missing openblas install in homebrew installation instructions
* Change charm++ instructions to i)built with shared libraries, and ii) build without ROMIO, which we don't use but which doesn't build for me for some reason I couldn't figure out
* Instructions now support building with shared libraries on and python bindings on (before they were off)
* Added a one liner at the end on how to install the python bindings in the user's python environment

Depends on #6036 

### Upgrade instructions

The instructions for building spectre on Apple Silicon have been updated, including enabling python bindings and building with shared libraries instead of static libraries.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: docs/Installation/InstallationOnAppleSilicon.md [hunk 1]
```diff
@@ -99,36 +97,21 @@ tar -xf blaze-3.8.tar.gz
...
 
 Next, clone, patch, and install charm++ v7.0.0.
 ```
 git clone https://github.com/UIUC-PPL/charm
 pushd charm
 git checkout v7.0.0
-./build LIBS multicore-darwin-arm8 --with-production -g3 -j
+git apply $SPECTRE_HOME/support/charm/v7.0.0.patch
+./build LIBS multicore-darwin-arm8 --with-production -g3 -j \
+--without-romio --with-shared
```
Comment (nilsvu) [inline docs/Installation/InstallationOnAppleSilicon.md]: The flag is `--build-shared` https://github.com/charmplusplus/charm?tab=readme-ov-file#building-dynamic-libraries

File: docs/Installation/InstallationOnAppleSilicon.md [hunk 2]
```diff
@@ -174,14 +155,13 @@ cmake \
 -D CMAKE_CXX_COMPILER=clang++ \
 -D CMAKE_Fortran_COMPILER=gfortran \
 -D CMAKE_BUILD_TYPE=Debug \
--D BUILD_SHARED_LIBS=OFF \
+-D BUILD_SHARED_LIBS=ON \
+-D BUILD_PYTHON_BINDINGS=ON \
```
Comment (nilsvu) [inline docs/Installation/InstallationOnAppleSilicon.md]: Pybindings are on by default now, just FYI

File: docs/Installation/InstallationOnAppleSilicon.md [hunk 3]
```diff
@@ -191,3 +171,8 @@ make -j10 unit-tests
 make -j10 test-executables
 ctest --output-on-failure -j10
 ```
+
+Optionally, to install the python bindings in your python environment,
+```
+pip install -e ${SPECTRE_HOME}/build/bin/python
```
Comment (nilsvu) [inline docs/Installation/InstallationOnAppleSilicon.md:177]: `make all-pybindings` before this

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: docs/Installation/InstallationOnAppleSilicon.md (hunk 1 above)
Comment (geoffrey4444) [reply docs/Installation/InstallationOnAppleSilicon.md]: 🤦‍♂️ thanks for catching this! Should have copied and pasted; the actual line I typed was of course --build-shared

File: docs/Installation/InstallationOnAppleSilicon.md (hunk 2 above)
Comment (geoffrey4444) [reply docs/Installation/InstallationOnAppleSilicon.md]: Good to know! Since they are on by default, I'll remove that line to streamline the docs

**Comment by geoffrey4444** [comment]
Thanks very much for taking a look at this, @nilsvu ! Because your changes were all very small one-liners, I squashed immediately. Ready for another look when you get a chance

**Review by nilsvu — APPROVED** [review]

**Pull request merged**